lib/src/log.dart

```diff
@@ -313,25 +315,72 @@ void exception(exception, [StackTrace? trace]) {
   }
 }
 
-/// Enables recording of log entries.
-void recordTranscript() {
-  _transcript = Transcript<_Entry>(_maxTranscript);
-}
-
-/// If [recordTranscript()] was called, then prints the previously recorded log
-/// transcript to stderr.
-void dumpTranscript() {
-  if (_transcript == null) return;
-
+/// Prints the recorded log transcript to stderr.
+void dumpTranscriptToStdErr() {
   stderr.writeln('---- Log transcript ----');
-  _transcript!.forEach((entry) {
+  _transcript.forEach((entry) {
     _printToStream(stderr, entry, showLabel: true);
   }, (discarded) {
     stderr.writeln('---- ($discarded discarded) ----');
   });
   stderr.writeln('---- End log transcript ----');
 }
 
+String _limit(String input, int limit) {
+  const snip = '[...]';
+  if (input.length < limit - snip.length) return input;
+  return '${input.substring(0, limit ~/ 2 - snip.length)}'
+      '$snip'
+      '${input.substring(limit)}';
+}
+
+/// Prints the previously recorded log transcript to stderr.
+void dumpTranscriptToFile(String path, String command, Entrypoint? entrypoint) {
+  final buffer = StringBuffer();
+  buffer.writeln('File containing information about the latest pub run.');
+  buffer.writeln(
+      'Before making this file public, make sure to remove any sensitive information!');
+  buffer.writeln();
+  buffer.writeln('Pub version: ${sdk.version}');
+  buffer.writeln(
+      'FLUTTER_ROOT: ${Platform.environment['FLUTTER_ROOT'] ?? '<not set>'}');
```

Comment:
Let's also dump `PUB_HOSTED_URL` that will save us a lot of pain...
And maybe `PUB_CACHE`?

Reply:
Great ideas!